From 55b4d4a3c9453b2c88ce3c1d7f6e83f6dac97e88 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Wed, 16 Feb 2022 00:20:23 +0800
Subject: [PATCH 46/50] Revert "drm: fix null-ptr-deref in
 drm_dev_init_release()"

This reverts commit 0499c863a8dbe22f1450e2cb3c14131956d91dd1.
---
 drivers/gpu/drm/drm_drv.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/gpu/drm/drm_drv.c b/drivers/gpu/drm/drm_drv.c
index 006e3b896cae..cd162d406078 100644
--- a/drivers/gpu/drm/drm_drv.c
+++ b/drivers/gpu/drm/drm_drv.c
@@ -577,7 +577,6 @@ static int drm_dev_init(struct drm_device *dev,
 			struct drm_driver *driver,
 			struct device *parent)
 {
-	struct inode *inode;
 	int ret;
 
 	if (!drm_core_init_complete) {
@@ -614,15 +613,13 @@ static int drm_dev_init(struct drm_device *dev,
 	if (ret)
 		return ret;
 
-	inode = drm_fs_inode_new();
-	if (IS_ERR(inode)) {
-		ret = PTR_ERR(inode);
+	dev->anon_inode = drm_fs_inode_new();
+	if (IS_ERR(dev->anon_inode)) {
+		ret = PTR_ERR(dev->anon_inode);
 		DRM_ERROR("Cannot allocate anonymous inode: %d\n", ret);
 		goto err;
 	}
 
-	dev->anon_inode = inode;
-
 	if (drm_core_check_feature(dev, DRIVER_RENDER)) {
 		ret = drm_minor_alloc(dev, DRM_MINOR_RENDER);
 		if (ret)
-- 
2.27.0

